### PR TITLE
Refactored the supported locale code out of the repositories

### PR DESCRIPTION
**Pull request by mamazu**
Seeing that most view repositories share the same code to determine the locale that the factory should use, I refactored this out into a provider class that can (and is) be tested.

### CODE REVIEW

**Comment by mamazu** [comment]
@lchrusciel what do you think?

File: spec/ViewRepository/ProductDetailsViewRepositorySpec.php [hunk 1]
```diff
@@ -130,39 +113,15 @@ function it_provides_product_view_by_code_in_given_locale(
...
-        ChannelInterface $channel,
-        LocaleInterface $locale
+        SupportedLocaleProviderInterface $supportedLocaleProvider,
+        ChannelInterface $channel
     ) {
-        $channel->getLocales()->willReturn(new ArrayCollection([$locale->getWrappedObject()]));
-        $locale->getCode()->willReturn('en_GB');
-
         $channelRepository->findOneByCode('WEB_GB')->willReturn($channel);
+        $supportedLocaleProvider->provide('de_DE', $channel);
```
Comment (lchrusciel) [inline spec/ViewRepository/ProductDetailsViewRepositorySpec.php]: Missing behaviour description

File: src/Provider/SupportedLocaleProvider.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            Assert::notNull($localeCode);
+        }
+
+        $this->assertLocaleSupport($localeCode, $channel->getLocales());
+
+        return $localeCode;
+    }
+
+    /**
+     * @param string $localeCode
```
Comment (lchrusciel) [inline src/Provider/SupportedLocaleProvider.php]: redundant

File: src/Provider/SupportedLocaleProvider.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+declare(strict_types=1);
+
+namespace Sylius\ShopApiPlugin\Provider;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Webmozart\Assert\Assert;
+
+class SupportedLocaleProvider implements SupportedLocaleProviderInterface
```
Comment (lchrusciel) [inline src/Provider/SupportedLocaleProvider.php]: Can we finalize this class?

File: spec/Provider/SupportedLocaleProviderSpec.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+use InvalidArgumentException;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Sylius\ShopApiPlugin\Provider\SupportedLocaleProvider;
+use Sylius\ShopApiPlugin\Provider\SupportedLocaleProviderInterface;
+
+final class SupportedLocaleProviderSpec extends ObjectBehavior
+{
+    function it_is_initializable()
```
Comment (lchrusciel) [inline spec/Provider/SupportedLocaleProviderSpec.php]: Missing void return types in spec

Comment (lchrusciel) [reply spec/Provider/SupportedLocaleProviderSpec.php]: I know, that's why I didn't put to much attention to this :)

Comment (mamazu) [reply spec/Provider/SupportedLocaleProviderSpec.php]: I will take it into account when creating further pull requests. :D

**Pull request merged**